From 643d6cb5d2dbb3fbf48a6aa7c831effd345f163d Mon Sep 17 00:00:00 2001
From: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
Date: Tue, 4 Sep 2018 18:24:36 +0900
Subject: [PATCH 355/909] media: rcar-vin: Add streaming count check when
 streming off

commit b762b33eeab851d5c4d43f95336c689769f56aa7 from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

When executing suspend to ram, in the case of abnormal termination,
media_pipeline_start is not executed and media_pipeline_stop only
may be executed in some cases.
In order to prevent it, this patch adds a check of stream count.

Signed-off-by: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/media/platform/rcar-vin/rcar-dma.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/media/platform/rcar-vin/rcar-dma.c b/drivers/media/platform/rcar-vin/rcar-dma.c
index 39e2755203c6..836d9578b745 100644
--- a/drivers/media/platform/rcar-vin/rcar-dma.c
+++ b/drivers/media/platform/rcar-vin/rcar-dma.c
@@ -1365,6 +1365,9 @@ static int rvin_set_stream(struct rvin_dev *vin, int on)
 	sd = media_entity_to_v4l2_subdev(pad->entity);
 
 	if (!on) {
+		if (vin->vdev.entity.stream_count <= 0)
+			return 0;
+
 		media_pipeline_stop(&vin->vdev.entity);
 		return v4l2_subdev_call(sd, video, s_stream, 0);
 	}
-- 
2.17.1

